src/components/Results.tsx

```diff
@@ -1,18 +1,100 @@
 import React, { MouseEventHandler } from "react";
+import { CATEGORY_NAMES } from "../constants";
+import { ScoreData } from "../types";
+import ResultsModal from "./ResultsByCategoryModal";
 
-const Results: React.FC<{
+interface QuestionCategories {
```

Comment:
It looks like the `freeCodeCampCorrect: number;` is missing because that category is listed in the `CATEGORY_NAMES` constant